rs/xnet/payload_builder/src/lib.rs

```diff
@@ -1559,21 +1527,20 @@ impl XNetClientImpl {
     /// most 1 idle connection per host.
     fn new(
         metrics_registry: &MetricsRegistry,
-        runtime_handle: runtime::Handle,
```

Comment:
Ignorant question: Why do we no longer use a specific runtime?

Reply:
I believe you implicitly use the same runtime as before because the runtime that polls the query method runs on the runtime as before defined here

https://github.com/dfinity/ic/blob/234ca3809b53192743763c0f7607e00da0ea2dce/rs/xnet/payload_builder/src/lib.rs#L1309

I don't have strong preference if we are explicit like before.

Reply:
I don't have a strong preference either. It's just that I have a vague memory from back when I was using `tokio` in my code that mixing multiple runtimes in a piece of code will lead to deadlocks (which is why we are passing this runtime everywhere instead of letting every module just use whatever the default runtime happens to be).

I may of course be misremembering, which is why I said "ignorant" above.

Reply:
I will be very surprised if you can do a deadlock unless you have blocking operations even if you mix runtimes.

None the less there is no mixing of runtimes even if we don't explicitly pass it.

Unless you run the xnet code with two different runtimes (which we don't) not sure there is a point in explicitly passing the runtime to the hyper client.

@alin-at-dfinity I hope you don't mind me resolving this comment.